[GTK] Update documentation

mcatanzaro opened the pull request:
#### a9f86aa9b53889b394c9b31435af884d802fda46
<pre>
[GTK] Update documentation
<a href="https://bugs.webkit.org/show_bug.cgi?id=251519">https://bugs.webkit.org/show_bug.cgi?id=251519</a>

Reviewed by Carlos Garcia Campos.

* Source/WebKit/UIProcess/API/glib/WebKitFaviconDatabase.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitFaviconDatabase.h.in:
* Source/WebKit/gtk/migrating-to-webkitgtk-6.0.md:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitFaviconDatabase.cpp:

Canonical link: <a href="https://commits.webkit.org/259967@main">https://commits.webkit.org/259967@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4902ca42999ba562fd03b7b221d678631fcd3261

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/106573 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/77/builds/15589 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/39382 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/115758 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/115399 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/110482 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/17084 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/6817 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/98768 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/112341 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/12981 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/39382 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/40539 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/94859 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/39382 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/82259 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/8816 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/39382 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/9358 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/5983 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/14977 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/39382 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6891 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/79/builds/10897 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebKit/gtk/migrating-to-webkitgtk-6.0.md:
```diff
@@ -88,12 +91,30 @@ APIs have been moved from [type@WebKit.WebContext] and [type@WebKit.WebsiteDataM
+[type@WebKit.FaviconDatabase] is now owned by [type@WebKit.WebsiteDataManager]
+instead of [type@WebKit.WebContext], and favicons are now enabled by default.
+Use [method@WebKit.WebsiteDataManager.set_favicons_enabled] if you do not need
+favicons.
```
Hm, maybe we should not enable favicon database by default, because favicons are not used for any purpose unless the application does something manually with WebKitFaviconDatabase....

carlosgcampos replied:
There's `WebKitWebView:favicon` property, you don't need to use `WebKitFaviconDatabase` at all, just monitor favicon property and use `webkit_web_view_get_favicon` to get the favicon. I agree with not enabling them by default, though.

mcatanzaro replied:
https://bugs.webkit.org/show_bug.cgi?id=251588

carlosgcampos commented on Source/WebKit/gtk/migrating-to-webkitgtk-6.0.md:
```diff
@@ -28,7 +28,11 @@ JavaScriptCore API (e.g. [type@JSC.Context] and [type@JSC.Object]) that is
 to interact with and manipulate the DOM instead, perhaps via
-[method@WebKit.WebView.run_javascript] or [method@JSC.ValueObject.invoke_method].
+[method@WebKit.WebView.evaluate_javascript] or
+[method@WebKit.WebView.call_async_javascript_function].
```
This is not correct, we are mixing ui and web process APIs here. `WebKitDOMDocument` is web process API, so the replacement would be the JSC GLib API. For example, to get the document you can:

```
JSCContext *jsContext = webkit_frame_get_js_context (frame);
JSCValue *jsDocument = jsc_context_get_value (jsContext, "document");
```

mcatanzaro replied:
Updated this a bit to clarify which process the APIs can be used in. Do you want me to additionally add that example? If so, I'll probably split it out into a separate section to keep this section small and focused.

mcatanzaro replied:
Well, I see you approved this PR, so I'll land now and we can follow up if desired.

mcatanzaro committed:
[GTK] Update documentation
https://bugs.webkit.org/show_bug.cgi?id=251519

Reviewed by Carlos Garcia Campos.

* Source/WebKit/UIProcess/API/glib/WebKitFaviconDatabase.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitFaviconDatabase.h.in:
* Source/WebKit/gtk/migrating-to-webkitgtk-6.0.md:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitFaviconDatabase.cpp:

Canonical link: https://commits.webkit.org/259967@main
